### PR TITLE
Included Export CSV for selection of Tags

### PR DESCRIPTION
**Pull request by swang1997**

### CODE REVIEW

File: export.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Locker,Address,Zone,Status,Name,Matrix Number,Course,Email
```
Comment (Tejasvas123) [inline export.csv]: add this to gitignore

File: export_selected.csv [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Locker,Zone,Status,Name
```
Comment (Tejasvas123) [inline export_selected.csv]: same as above

File: src/main/java/duke/logic/commands/ExportLockerSelectCommand.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import duke.models.LockerList;
+import duke.models.locker.Locker;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+public class ExportLockerSelectCommand extends Command {
+
+    @Override
+    public void execute(LockerList lockerList, Ui ui, Storage storage) throws DukeException {
+        ui.exportSelect();
```
Comment (Tejasvas123) [inline src/main/java/duke/logic/commands/ExportLockerSelectCommand.java]: why are you passing ui.readCommand()?

Comment (Tejasvas123) [reply src/main/java/duke/logic/commands/ExportLockerSelectCommand.java]: the read command is already being executed in duke.java

File: src/main/java/duke/parser/ExportLockerSelectCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package duke.parser;
+
+import duke.logic.commands.ExportLockerSelectCommand;
+
+public class ExportLockerSelectCommandParser {
+
+    /**
+     * This function is used to parse the user input for exporting the details as a CSV file.
+     * @param arguments this is the user input string for the tags that the user wants
```
Comment (Tejasvas123) [inline src/main/java/duke/parser/ExportLockerSelectCommandParser.java]: the actual parsing should be done here. Like basically in your case you should parse what all fields are being included into export. Have a look at some examples like AddCommandParser/EditCommandParser to understand what i mean

File: src/main/java/duke/parser/Parser.java [hunk 5]
```diff
@@ -24,11 +24,15 @@
     private static final Pattern GENERAL_COMMAND_FORMAT =
             Pattern.compile("(?<commandType>\\S+)(?<arguments>.*)");
 
+    private static final String LIST_COMMAND = "list";
+    private static final String EXIT_COMMAND = "bye";
+    private static final String EXPORT_LOCKER_COMMAND = "export";
+    private static final String EXPORT_SELECT_COMMAND = "exports";
```
Comment (Tejasvas123) [inline src/main/java/duke/parser/Parser.java]: if i am not wrong i removed all this in my latest PR. The one about code cleaning

File: src/main/java/duke/storage/ExportCsv.java [hunk 6]
```diff
@@ -49,7 +49,8 @@ public static void exportLockers(List<Locker> lockerList) throws DukeException {
                 details[THIRD_COLUMN] = l.getZone().getZone();
                 details[FORTH_COLUMN] = l.getTag().getTagName();
 
-                if (details[THIRD_COLUMN].equals("in-use")) {
+                if (details[FORTH_COLUMN].equals("in-use")) {
+                    
                     details[FIFTH_COLUMN] = l.getUsage().get().getStudent().getName().getName();
                     details[SIXTH_COLUMN] = l.getUsage().get().getStudent().getMatricNumber().getMatricId();
                     details[SEVENTH_COLUMN] = l.getUsage().get().getStudent().getMajor().getCourse();
```
Comment (Tejasvas123) [inline src/main/java/duke/storage/ExportCsv.java]: how about exporting rental dates also? or like no-need for that?

File: src/main/java/duke/storage/ExportSelection.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ExportSelection {
+    private static final String CSV_OUTPUT_PATH = "export_selected.csv";
```
Comment (Tejasvas123) [inline src/main/java/duke/storage/ExportSelection.java]: i was thinking of using only one file.  Like just one export.csv

File: src/main/java/duke/ui/Ui.java [hunk 8]
```diff
@@ -13,6 +13,9 @@ public String readCommand() {
         return sc.nextLine();
     }
 
+    public String read() {
```
Comment (Tejasvas123) [inline src/main/java/duke/ui/Ui.java]: why is this needed?

File: src/main/java/duke/ui/Ui.java [hunk 9]
```diff
@@ -129,6 +132,30 @@ public void showNoAvailableLockers() {
         printSpaces(" There are no available lockers at the moment. ");
         printSpaces(" Unfortunately I will have to terminate the subscription of the student");
     }
+
```
Comment (Tejasvas123) [inline src/main/java/duke/ui/Ui.java]: remove this snoozetask function

File: src/main/java/duke/ui/Ui.java [hunk 10]
```diff
@@ -193,6 +220,30 @@ public void exportMessage() {
...
+    /**
+     * This function tells the user that SpongeBob has exported the CSV file.
+     */
+    public void exportSelect() {
+        printSpaces("Noted. I have exported the selected details to a new CSV file");
+        printDash();
+    }
+
+
+    /**
```
Comment (Tejasvas123) [inline src/main/java/duke/ui/Ui.java]: remove functions keywordPrint and remindersList

**Review by Tejasvas123 — DISMISSED** [review]

File: export.csv (hunk 1 above)
Comment (swang1997) [reply export.csv]: added

File: export_selected.csv (hunk 2 above)
Comment (swang1997) [reply export_selected.csv]: removed export_selected.csv. Both exports will share same csv file.

File: src/main/java/duke/logic/commands/ExportLockerSelectCommand.java (hunk 3 above)
Comment (swang1997) [reply src/main/java/duke/logic/commands/ExportLockerSelectCommand.java]: edited in parser

File: src/main/java/duke/parser/ExportLockerSelectCommandParser.java (hunk 4 above)
Comment (swang1997) [reply src/main/java/duke/parser/ExportLockerSelectCommandParser.java]: edited

File: src/main/java/duke/parser/Parser.java (hunk 5 above)
Comment (swang1997) [reply src/main/java/duke/parser/Parser.java]: removed and cleaned

File: src/main/java/duke/storage/ExportCsv.java (hunk 6 above)
Comment (swang1997) [reply src/main/java/duke/storage/ExportCsv.java]: will try doing it....

File: src/main/java/duke/storage/ExportSelection.java (hunk 7 above)
Comment (swang1997) [reply src/main/java/duke/storage/ExportSelection.java]: fixed

File: src/main/java/duke/ui/Ui.java (hunk 8 above)
Comment (swang1997) [reply src/main/java/duke/ui/Ui.java]: removed

File: src/main/java/duke/ui/Ui.java (hunk 9 above)
Comment (swang1997) [reply src/main/java/duke/ui/Ui.java]: removed

File: src/main/java/duke/ui/Ui.java (hunk 10 above)
Comment (swang1997) [reply src/main/java/duke/ui/Ui.java]: removed

File: src/main/java/duke/parser/ExportLockerSelectCommandParser.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+                if (str.equals("status")) {
+                    checkStatus = 1;
+                }
+            }
+            if (checkStatus == 0) {
+                for (String str : newArgs) {
+                    if (str.equals("name")) {
+                        throw new DukeException(ExportLockerSelectCommand.MISSINGSTATUS_FORMAT);
+                    }
+                    if (str.equals("matrix")) {
```
Comment (Tejasvas123) [inline src/main/java/duke/parser/ExportLockerSelectCommandParser.java]: should be matricid /student id?

File: src/main/java/duke/storage/Storage.java [hunk 12]
```diff
@@ -1,5 +1,6 @@
 package duke.storage;
 
+import duke.Duke;
```
Comment (Tejasvas123) [inline src/main/java/duke/storage/Storage.java]: unused class?

File: src/main/java/duke/ui/Ui.java [hunk 13]
```diff
@@ -193,6 +192,14 @@ public void exportMessage() {
         printDash();
     }
 
+    /**
+     * This function tells the user that SpongeBob has exported the CSV file.
+     */
+    public void exportSelect() {
+        printSpaces("Noted. I have exported the selected details to a new CSV file");
```
Comment (Tejasvas123) [inline src/main/java/duke/ui/Ui.java]: just write the name of the file instead of writing csv files

**Review by Tejasvas123 — COMMENTED** [review]
Suggested some minor changes. Great job otherwise!

File: src/main/java/duke/ui/Ui.java (hunk 13 above)
Comment (Tejasvas123) [reply src/main/java/duke/ui/Ui.java]: does it work as intended? like no errors?

File: src/main/java/duke/storage/StorageManager.java [hunk 14]
```diff
@@ -32,4 +37,9 @@ public void exportAsCsv(LockerList lockerList) throws DukeException {
         requireNonNull(lockerList);
         writeToCsv.exportLockers(lockerList.getAllLockers());
     }
+
+    public void exportSelection(LockerList lockerList, String input) throws DukeException {
```
Comment (Tejasvas123) [inline src/main/java/duke/storage/StorageManager.java]: forgot to add override?

**Review by Zhubo225 — DISMISSED** [review]

**Review by Tejasvas123 — APPROVED** [review]
Approved! Good job!
